emission/tests/funcTests/TestFunctionTiming.py

```diff
@@ -35,207 +17,114 @@ def setUpClass(self):
         """
         Test execution and timing of a single function.
         """
         def sample_function():
-            logging.info("Executing sample_function")
+            logging.debug("Executing sample_function")
             time.sleep(2)  # Simulate processing time
             return True
 
-        with test_logger("test_single_function_timing"):
-            self.run_with_timing("sample_function", sample_function)
-
-    def test_multiple_functions_timing(self):
-        """
-        Test execution and timing of multiple functions within a single test.
-        Ensures that each function's timing is recorded separately.
-        """
-        def function_one():
-            logging.info("Executing function_one")
-            time.sleep(1)
-            return True
-
-        def function_two():
-            logging.info("Executing function_two")
-            time.sleep(1.5)
-            return True
-
-        functions = [function_one, function_two]
+        with ect.Timer() as timer:
+            sample_function()
 
-        with test_logger("test_multiple_functions_timing"):
-            for func in functions:
-                self.run_with_timing(func.__name__, func)
+        esdsq.store_dashboard_time("sample_function", timer)
 
+        # verify that an entry was stored correctly
+        self.verify_stats_entries([
+            ("stats/dashboard_time", "sample_function", timer.elapsed_ms)
+        ])
 
-    def test_code_block_instrumentation(self):
+    def test_multiple_functions_timing(self):
         """
-        Test instrumentation of a specific code block using the Timer context manager.
-        Ensures that timing information for the code block is recorded correctly.
+        Test execution and timing of 2 functions, where we record and validate
+        the time taken for: (i) function_one, (ii) function_two, and (iii) both functions together.
         """
-        block_name = "sample_code_block"
+        def function_one(): return time.sleep(1)
+        def function_two(): return time.sleep(1.5)
 
-        def code_block():
-            logging.info("Executing sample_code_block with simulated operations")
-            logging.info("Step 1: Data retrieval from mock service")
-            time.sleep(0.5)  # Simulate network delay
-            data = {"user_id": 123, "action": "login"}
-            logging.info(f"Data retrieved: {data}")
+        with ect.Timer() as timer_both:
+            with ect.Timer() as timer_one:
+                function_one()
+            esdsq.store_dashboard_time("function_one", timer_one)
 
-            logging.info("Step 2: Data processing")
-            time.sleep(1)  # Simulate processing time
-            processed_data = {k: str(v).upper() for k, v in data.items()}
-            logging.info(f"Processed data: {processed_data}")
+            with ect.Timer() as timer_two:
+                function_two()
+            esdsq.store_dashboard_time("function_two", timer_two)
```

Comment:
You might have nested timers. This will be necessary if you want to time the "stages" of some larger operation, as well as the total time of the operation.